### DISCUSSION TITLE
Inlining call frames in `BEFORE_WITH`/`BEFORE_ASYNC_WITH`

### DISCUSSION THREAD

**Discussion by Fidget-Spinner** [faster-cpython/ideas#398] [Ideas]
Recently I noticed a lot of use patterns like this (especially in aiohttp):
```
async def f():
    async with req.get():
        ...
```
I wonder if we should try making `__aenter__` and `__aexit__` stackless in the `BEFORE_ASYNC_WITH` instruction and its non asnyc counterpart `BEFORE_WITH`. This will work similarly to `__getitem__` in `BINARY_SUBSCR`. This will likely show zero speedup in pyperformance, but if may move something like the aiohttp benchmark in Pyston.

Asking around before I invest time into this because it may not be worth the effort.

**Comment by markshannon** [comment]
I doubt it is worth it adding the complexity.

I'd rather break apart `BEFORE_ASYNC_WITH` and `BEFORE_WITH` into lower level instructions:
```
COPY 1
LOAD_SPECIAL __enter__
SWAP 2
LOAD_SPECIAL __exit__
SWAP 2
CALL 0
```
